src/App.tsx

```diff
@@ -1,61 +1,122 @@
+import { useState } from 'react';
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { TodoWithUser } from './types/TodoWithUser';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const combined = todosFromServer.map(todo => {
+  const user = usersFromServer.find(
+    userServer => userServer.id === todo.userId,
+  );
+
+  return { ...todo, user };
+});
 
 export const App = () => {
+  const [todos, setTodosFromServer] = useState([...combined]);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [newId, setNewId] = useState(0);
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+
+  const reset = () => {
+    setNewTodoTitle('');
+    setErrorTitle(false);
+    setErrorUser(false);
+  };
+
+  const handleAddTodo = () => {
+    if (newTodoTitle === '') {
+      setErrorTitle(true);
+    }
+
+    if (newId === 0) {
+      setErrorUser(true);
+    }
+
+    if (newTodoTitle === '' || newId === 0) {
+      return;
+    }
+
+    if (newTodoTitle.trim() === '') {
+      setErrorTitle(true);
+
+      return;
+    }
+
+    const newTodo: TodoWithUser = {
+      id: todosFromServer.length + 1,
+      title: newTodoTitle,
+      completed: false,
+      userId: newId,
+      user: {
+        email: usersFromServer[newId - 1].email,
+        id: newId,
+        name: usersFromServer[newId - 1].name,
+        username: usersFromServer[newId - 1].username,
+      },
+    };
```

Comment:
DRY - don't repeat yourself
```suggestion
    const user = usersFromServer[newId - 1];

    const newTodo: TodoWithUser = {
      id: todosFromServer.length + 1,
      title: newTodoTitle,
      completed: false,
      userId: newId,
      user: {
        email: user.email,
        id: newId,
        name: user.name,
        username: user.username,
      },
    };
```